source/particle/interpolator/quadratic_least_squares.cc

```diff
@@ -251,34 +575,63 @@ namespace aspect
             {
               prm.enter_subsection("Quadratic least squares");
               {
-                use_global_values_limiter = prm.get_bool("Use limiter");
-                if (use_global_values_limiter)
+                const Postprocess::Particles<dim> &particle_postprocessor =
+                  this->get_postprocess_manager().template get_matching_postprocessor<const Postprocess::Particles<dim>>();
+                const auto &particle_property_information = particle_postprocessor.get_particle_world().get_property_manager().get_data_info();
+                const unsigned int n_property_components = particle_property_information.n_components();
+                const unsigned int n_internal_components = particle_property_information.get_components_by_field_name("internal: integrator properties");
+
+                const std::vector<std::string> quadratic_least_squares_limiter_split = Utilities::split_string_list(prm.get("Use quadratic least squares limiter"));
+                std::vector<bool> quadratic_least_squares_limiter_parsed;
+                if (quadratic_least_squares_limiter_split.size() == 1)
+                  {
+                    quadratic_least_squares_limiter_parsed = std::vector<bool>(n_property_components - n_internal_components, internal::string_to_bool(quadratic_least_squares_limiter_split[0]));
+                  }
+                else if (quadratic_least_squares_limiter_split.size() == n_property_components - n_internal_components)
+                  {
+                    for (const auto &component: quadratic_least_squares_limiter_split)
+                      quadratic_least_squares_limiter_parsed.push_back(internal::string_to_bool(component));
+                  }
+                else
+                  {
+                    AssertThrow(false, ExcMessage("The size of 'Use quadratic least squares limiter' should either be 1 or the number of particle properties"));
+                  }
+                for (unsigned int i = 0; i < n_internal_components; ++i)
+                  quadratic_least_squares_limiter_parsed.push_back(false);
+                use_quadratic_least_squares_limiter = ComponentMask(quadratic_least_squares_limiter_parsed);
+
+                limiter_enabled_for_at_least_one_property = (use_quadratic_least_squares_limiter.n_selected_components(n_property_components) != 0);
```

Comment:
Oh, I see now why you stored this, you dont want to access the postprocessor to compute `n_components` again above. Ok, then. Keep the variable.